### ISSUE TITLE
db load API does not load trigger statement 

### ISSUE THREAD

**Issue by gilwo** [rqlite/rqlite#246]
Hi,

**tl;dr:**
as the titles say `db/load` api fail to load trigger statements.
more info: manually posting was fine, using the CLI is fine as well, working in sqlite as expected.

**extended info:**
i used the method of dumping a sqlite db into file and then loading it to rqlite usng db/load api.
everything beside triggers were loaded 

my `test.dump` file content:
```
PRAGMA foreign_keys=OFF;
BEGIN TRANSACTION;
CREATE TABLE foo (id integer primary key asc, name text);
INSERT INTO "foo" VALUES(1,'bob');
INSERT INTO "foo" VALUES(2,'alice');
INSERT INTO "foo" VALUES(3,'eve');
CREATE TABLE bar (nameid integer, age integer);
INSERT INTO "bar" VALUES(1,44);
INSERT INTO "bar" VALUES(2,46);
INSERT INTO "bar" VALUES(3,8);
CREATE VIEW foobar as select name as Person, Age as age from foo inner join bar on foo.id == bar.nameid;
CREATE TRIGGER new_foobar instead of insert on foobar begin insert into foo (name) values (new.Person); insert into bar (nameid, age) values ((select id from foo where name == new.Person), new.Age); end;
COMMIT;
```

using the load API
```
curl -XPOST  "localhost:4001/db/load?pretty&timing" --data-binary @test.dump
```
tried to modify the view which will fire the trigger :
```
curl -XPOST 'localhost:4001/db/execute?pretty&timings' -H "Content-Type: application/json" -d "[\"insert into foobar values ('jason', 16);\"]"
```
resulted in :
```
{
    "results": [
        {
            "error": "cannot modify foobar because it is a view"
        }
    ],
    "time": 0.045988233
}
```

then checked rqlite tables :
```
curl -G 'localhost:4001/db/query?timings&pretty' --data-urlencode 'q=SELECT * FROM sqlite_master'
```

result was :
```
{
    "results": [
        {
            "columns": [
                "type",
                "name",
                "tbl_name",
                "rootpage",
                "sql"
            ],
            "types": [
                "text",
                "text",
                "text",
                "integer",
                "text"
            ],
            "values": [
                [
                    "table",
                    "foo",
                    "foo",
                    2,
                    "CREATE TABLE foo (id integer primary key asc, name text)"
                ],
                [
                    "table",
                    "bar",
                    "bar",
                    3,
                    "CREATE TABLE bar (nameid integer, age integer)"
                ],
                [
                    "view",
                    "foobar",
                    "foobar",
                    0,
                    "CREATE VIEW foobar as select name as Person, Age as age from foo inner join bar on foo.id == bar.nameid"
                ]
            ],
            "time": 9.9511e-05
        }
    ],
    "time": 0.00011753100000000001
}
```

manually posting the trigger :
```
curl -XPOST 'localhost:4001/db/execute?pretty&timings' -H "Content-Type: application/json" -d "[\"CREATE TRIGGER new_foobar instead of insert on foobar begin insert into foo (name) values (new.Person); insert into bar (nameid, age) values ((select id from foo where name == new.Person), new.Age); end;\"]"
```
resulted in:
```{
    "results": [
        {
            "last_insert_id": 3,
            "rows_affected": 1,
            "time": 0.00023702300000000002
        }
    ],
    "time": 0.073914539
}
```

now trying again :
```
curl -XPOST 'localhost:4001/db/execute?pretty&timings' -H "Content-Type: application/json" -d "[\insert into foobar values ('jason', 16);\"]"
```
resulted in :
```
{
    "results": [
        {
            "last_insert_id": 3,
            "time": 0.0001294
        }
    ],
    "time": 0.041909619
}
```
and the trigger is there now.

**Comment by otoolep** [comment]
Thanks for the report @gilwo -- I see what is going on. It's due to the very simple parser in the rqlite.

It might take a little while to fix this -- you should load triggers manually for now, as you are doing.

**Comment by otoolep** [comment]
Actually, this looks easier to fix that I though. I'm working on it in https://github.com/rqlite/rqlite/pull/247

**Comment by otoolep** [comment]
Fixed by https://github.com/rqlite/rqlite/pull/247

**Comment by otoolep** [comment]
@gilwo -- try the latest build from master, and tell me if all looks good now.

**Comment by otoolep** [comment]
Actually, this doesn't look quite right yet, re-opening.

**Comment by otoolep** [comment]
It's all good, you must set the right `Content-type` header during the `curl` command. See https://github.com/rqlite/rqlite/blob/master/doc/RESTORE_FROM_SQLITE.md for more details.

**Comment by gilwo** [comment]
i just tested it and something is not working very good.
my project db structure is quite big and it appear that for large triggers the dump still wont go in  using the db/load api.
the different is that now the transaction is failing as it gives a syntax error.
its quite complex to give an example to this as i cant share the db structure itself.
is there a way to open logs or run rqlited in debug mode ? 
thanks

**Comment by otoolep** [comment]
It is possible @gilwo that very large dumps may not go through. 

Can you show me the syntax error?

**Comment by otoolep** [comment]
There is no explicit debug mode, apart from what you see in the logs. If you wish to share you database dump file with just me, you could e-mail me a compressed file or link to it.
